src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -165,8 +165,24 @@ private async Task<FormulaValue> TryHandleSet(CallNode node, EvalVisitorContext
             var arg0 = node.Args[0];
             var arg1 = node.Args[1];
 
-            var newValue = await arg1.Accept(this, context).ConfigureAwait(false);
-
+            var newValue = await arg1.Accept(this, context).ConfigureAwait(false);
+
+            if (arg0.IRContext.IsMutation && arg0 is RecordFieldAccessNode rfan)

```

Comment:
>IRContext [](http://example.com/codeflow?start=21&length=9)

Some of these look like previous changes? Is this rebased on latest?

Reply:
It is.